services/mailservice/inbucket.go

```diff
@@ -99,7 +100,13 @@ func GetMessageFromMailbox(email, id string) (JSONMessageInbucket, error) {
 	if err != nil {
 		return record, err
 	}
-	defer emailResponse.Body.Close()
+	defer func() {
+		// To reuse same tcp connection
+		// Response body is read and closed.
+
+		io.Copy(ioutil.Discard, emailResponse.Body)
```

Comment:
Well, we are creating a new HTTP client every time, so actually this code doesn't give us much benefit. To actually reuse the connection, we need to change the `get` function to use `http.Get` rather than construct the client from scratch.

Reply:
ok I will update the get function, just to be sure you are asking about this `req, err := http.NewRequest("GET", url, nil)` to `http.Get(url)` @agnivade

Reply:
That's correct.

Reply:
can you check now?